lib/Conversion/SCFToCalyx/SCFToCalyx.cpp

```diff
@@ -698,6 +753,52 @@ LogicalResult BuildOpGroups::buildOp(PatternRewriter &rewriter,
           yieldOp->getOpOperands());
   getState<ComponentLoweringState>().setWhileLoopLatchGroup(whileOpInterface,
                                                             assignGroup);
+  }
+  else if (auto ifOp = dyn_cast<scf::IfOp>(yieldOp->getParentOp())) {
+
+    ScfIfOp scfIfOp(ifOp);
```

Comment:
I don't think this is used, is it intended to be used? I think in general, we probably don't need the interfaces you've added here, if there is only one thing implementing the interface. The reason we have interfaces for the while op is because we want to generically handle both scf::WhileOp and our own loopschedule::WhileOp.

Reply:
I think you are right, adding an interface is redundant for `if`